### ISSUE TITLE
Link header full compliancy with RFC 5988

### ISSUE THREAD

**Issue by Sir4ur0n** [spring-projects/spring-hateoas#710]
Hello,

Spring HATEOAS seems to have at least partial support for web linking (as shown by class `org.springframework.hateoas.Links`) however its pattern does not support valid RFC 5988 headers.

Example of header from the RFC: `<http://example.com/TheBook/chapter2>; rel="previous";title="previous chapter"`

This does not match `org.springframework.hateoas.Links#LINK_HEADER_PATTERN` because of the space between `;` and `rel`

Some applications (e.g. JHipster) generate a Link header with space, thus `org.springframework.hateoas.Links#valueOf` does not parse its header.

I admit I did not check if there are any other inconsistencies with the RFC, sorry about that, it's not exactly the most interesting RFC to read 😄 

Here's the link to RFC 5988 about Web linking (Link header): https://tools.ietf.org/html/rfc5988

Bonus question: shouldn't `Links` and `Link` belong to spring-web instead of spring-hateoas? This way, there could be a method `public List<Link> getLinks()` or `public Links getLinks()` in `org.springframework.http.HttpHeaders`.

Cheers

**Comment by gregturn** [comment]
Inside `LinkUnitTest`, I just coded two unit tests based on these examples...

```
	@Test
	public void parsesRFC5988HeaderIntoLink() {

		assertThat(Link.valueOf("<http://example.com/TheBook/chapter2>; rel=\"previous\";title=\"previous chapter\""))
			.isEqualTo(new Link("http://example.com/TheBook/chapter2").withRel("previous").withTitle("previous chapter"));
		assertThat(Link.valueOf("<http://example.com/TheBook/chapter2>;rel=\"previous\";title=\"previous chapter\""))
			.isEqualTo(new Link("http://example.com/TheBook/chapter2").withRel("previous").withTitle("previous chapter"));

		assertThat(Links.valueOf("<http://example.com/TheBook/chapter2>; rel=\"previous\";title=\"previous chapter\""))
			.containsExactlyInAnyOrder(new Link("http://example.com/TheBook/chapter2").withRel("previous").withTitle("previous chapter"));
		assertThat(Links.valueOf("<http://example.com/TheBook/chapter2>;rel=\"previous\";title=\"previous chapter\""))
			.containsExactlyInAnyOrder(new Link("http://example.com/TheBook/chapter2").withRel("previous").withTitle("previous chapter"));
		...
}
```
...and it passed green. Looking at the pattern in `Links`, I see `;\\s*` in the middle, which says "zero more more spaces", so it would be none or one. And these unit tests show that to work, whether for a single link or multiples.

Are you using a different version of Spring HATEOAS? Because these fixes were applied in #686.

As for Spring HATEOAS being inside spring-web, it's important to realize that Spring HATEOAS came much later than spring-web. It needs a faster turn around cycle before it could possibly be folded into Spring Framework itself.

**Comment by Sir4ur0n** [comment]
Hi @gregturn 

You're right, I thought I had a recent version of Spring HATEOAS but apparently not, it's my bad :'(
I see it's been fixed in recent versions (I dug up and found #686).

Sorry 😞
